src/seqn/seqn.lisp

```diff
@@ -141,4 +141,57 @@ For a less formal description consult the specs file"
     (seqn-lt          (list 1 0))
     (otherwise (geb.utils:subclass-responsibility x))))
 
-
+(defmethod gapply ((morphism <seqn>) vector)
+  "Takes a list of vectors of natural numbers and gives out their evaluations.
+Currently does not correspond directly to the intended semantics but
+is capable of succesfully evaluating all compiled terms"
+  (etypecase-of seqn morphism
+    (id               vector)
+    (composition      (gapply (mcar morphism)
+                              (gapply (mcadr morphism) vector)))
+    (parallel-seq     (append (gapply (mcar morphism)
+                                      (subseq vector
+                                              0 (length (dom (mcar morphism)))))
+                              (gapply (mcadr morphism)
+                                      (subseq vector
+                                              (length (dom (mcar morphism)))))))
+    (fork-seq         (append vector vector))
+    (drop-nil         (list 0))
+    (remove-right     (butlast vector))
+    (remove-left      (cdr vector))
+    (drop-width       vector)
+    (inj-length-left  (append vector
+                              (mapcar (lambda (x) 0)
+                                      (mcadr morphism))))
+    (inj-length-right (append (mapcar (lambda (x) 0)
+                                      (mcar morphism))
```

Comment:
sbcl will yell at you for unused x. just use the `make-list` I used in my previous comments for these. would read much better. Or in this case use the `pad-with-zeros` I mentioned before.

Maybe make the API of `pad-with-zeros` take a keyword for prepend or postpend, doing the more useful one by default.